File: app/src/main/java/s/yarlykov/izisandbox/recycler_and_swipes/smart_adapter/sand_box/java/BaseItem.java
package s.yarlykov.izisandbox.recycler_and_swipes.smart_adapter.sand_box.java;

import androidx.recyclerview.widget.RecyclerView;

public class BaseItem<H extends RecyclerView.ViewHolder> {

    /**
     * Next item in ItemList
     */
    public BaseItem<H> nextItem;

    /**
     * Previous item in ItemList
     */
    public BaseItem<H> previousItem;

}

File: app/src/main/java/s/yarlykov/izisandbox/recycler_and_swipes/smart_adapter_v2/item/BaseItem.kt
package s.yarlykov.izisandbox.recycler_and_swipes.smart_adapter_v2.item

import androidx.recyclerview.widget.RecyclerView
import s.yarlykov.izisandbox.recycler_and_swipes.smart_adapter_v2.controller.BaseItemController

/**
 * Базовый контейнер для элемента модели. Это не элемент модели, а именно контейнер.
 * Он ничего не знает о том как будет отображаться его элемент на экране.
 * Всю работу по отображению он делегирует контроллеру.
 */
open class BaseItem<H : RecyclerView.ViewHolder>(val controller: BaseItemController<H, BaseItem<H>>)